terraform/main.tf

```diff
@@ -0,0 +1,144 @@
+
+provider "aws" {
+  region = var.region
+  default_tags {
+    tags = var.default_tags_tf
+  }
+}
+
+
+resource "aws_kms_key" "this" {
+  description = "KMS key for MSK SCRAM credentials"
+  tags        = var.tags
+}
+
+resource "aws_kms_alias" "this" {
+  name          = "alias/${var.identifier}-msk-scram"
+  target_key_id = aws_kms_key.this.key_id
+}
+
+resource "aws_cloudwatch_log_group" "this" {
+  count = var.logging_info != null ? 1 : 0
+
+  name              = "${var.identifier}-msk-broker-logs"
+  retention_in_days = var.logging_info.broker_logs.cloudwatch_logs.retention_in_days
+  tags              = var.tags
+}
+
+resource "aws_msk_configuration" "this" {
+  name              = "${var.identifier}-${replace(var.kafka_version, ".", "-")}"
+  kafka_versions    = [var.kafka_version]
+  server_properties = var.server_properties
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_secretsmanager_secret" "this" {
+  for_each   = var.scram_users
+  name       = "AmazonMSK_${var.identifier}-${each.key}"
+  tags       = var.tags
+  kms_key_id = aws_kms_key.this.key_id
+}
+
+resource "aws_secretsmanager_secret_version" "this" {
+  for_each  = var.scram_users
+  secret_id = aws_secretsmanager_secret.this[each.key].id
+  # enforce the order of the keys in the JSON object
+  secret_string = jsonencode({
+    for k in sort(keys(each.value)) : k => each.value[k]
+  })
+}
+
+resource "aws_secretsmanager_secret_policy" "this" {
+  for_each   = var.scram_users
+  secret_arn = aws_secretsmanager_secret.this[each.key].arn
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Sid    = "AWSKafkaResourcePolicy"
+        Effect = "Allow"
+        Principal = {
+          Service = "kafka.amazonaws.com"
+        }
+        Action   = "secretsmanager:getSecretValue"
+        Resource = aws_secretsmanager_secret.this[each.key].arn
+      }
+    ]
+  })
+}
+
+resource "aws_msk_scram_secret_association" "this" {
+  count = length(var.scram_users) > 0 && var.client_authentication != null && var.client_authentication.sasl.scram ? 1 : 0
```

Comment:
terraform `&&` is not short circuit, if `ar.client_authentication` is `null`, then `var.client_authentication.sasl.scram` will throw error

Reply:
Changed in b251a62